### ISSUE TITLE
[QUESTION] How to avoid duplicate errors?

### ISSUE THREAD

**Issue by Micu22** [WordPress/WordPress-Coding-Standards#1961]
Very ofter I get a report that contains something like this:

26 | ERROR   | [x] Space after opening control structure is required
26 | ERROR   | [x] No space before opening parenthesis is prohibited
26 | ERROR   | [x] Expected 1 space(s) after IF keyword; 0 found

Clearly these three lines mean the same thing. Is there a way to disable 2 of them?

Also I often come across errors like this: 
22 | ERROR   | [x] Whitespace found at end of line
Is it really important? What is the difference between a space at end of line and no space? If there is little to none, then how can I disable this error too?

**Comment by jrfnl** [comment]
Please fill in the complete bug report/question template. Without a code sample or knowing which versions of PHPCS/WPCS you are using, it is  lot more difficult for us to answer your questions.

**Comment by Micu22** [comment]
Code sample:
```
if( ! is_single() ) {
	return $var;
}
```
phpcs version: 3.5.8
wpcs version: (i think) 2.3, I have installed it with 
> composer create-project wp-coding-standards/wpcs --no-dev
two days ago.

**Comment by dingo-d** [comment]
I would be cool if you told us which coding standard you're using too 😄 

The error messages are certainly not the same. The first one is telling you that you need a space after the control structure (`if (`). The second one says that you must have space before opening a parenthesis (ties into the control structure, but is a totally different sniff), and the last one says what you need to fix (again, a separate sniff).

Without the standards used, and full error messages that will specify which sniffs are throwing the errors it's just guessing at this point.

But, even without that, it's not an error, since those are different sniffs 🙂

**Comment by Micu22** [comment]
I am using WordPress coding standard. 
I have this enabled:
```
<rule ref="WordPress-Docs"/>
<rule ref="WordPress-Extra"/>
```
I understand it's not an error, that is why I marked this as a question not an issue.

What about this? `22 | ERROR | [x] Whitespace found at end of line` how to turn this off? I don't care about some single spaces at the end of lines.

btw. are you using Sublime Text 3? I am looking for someone who got phpcs package to work in Sublime, I have no luck.

**Comment by dingo-d** [comment]
You could turn that off in your custom standard with (I think it's coming from this standard)

```xml
<rule ref="Squiz.WhiteSpace.SuperfluousWhitespace">
  <properties>
    <property name="ignoreBlankLines" value="true"/>
  </properties>
</rule>
```

As for the logic of removing it you can check this answer: https://softwareengineering.stackexchange.com/a/121560

Unfortunately, I'm not using ST3 anymore, but when I used it (installed from a project) I didn't have any issues 🤷‍♂️ 

DId you check the wiki: https://github.com/WordPress/WordPress-Coding-Standards/wiki/Setting-up-WPCS-to-work-in-Sublime-Text ?

**Comment by Micu22** [comment]
Thanks.
